bin/employee_permissions/tool_permission/code_org_employee_permission.rb

```diff
@@ -0,0 +1,53 @@
+class CodeOrgEmployeePermission
+  def initialize(opts)
+    @opts = opts
+  end
+
+  def id
+    return @opts[:id]
+  end
+
+  def admin?
+    return @opts[:is_admin]
+  end
+
+  def name
+    return @opts[:name]
+  end
+
+  def email
+    return @opts[:email]
+  end
+
+  def search_key
+    return @opts[:search_key]
+  end
+
+  def remove(dry_run=false)
+    raise "Not implemented"
+  end
+
+  def self.get_all_members_impl
+    raise "Not implemented"
+  end
+
+  def self.get_all_members
+    raw_members = get_all_members_impl
+    members = []
+    raw_members.each do |raw_member|
+      new_member = new(raw_member)
+      members << new_member
+    end
+    members
+  end
+
+  def self.get_member_from_search_key(search_key)
+    results = []
+    get_all_members.each do |member|
+      if member.search_key == search_key
+        results << member
+      end
+    end
```

Comment:
Tiny nit: this could just be

```suggestion
    results = get_all_members.select do |member|
      member.search_key == search_key
    end
```

Reply:
and Ruby encourages doing this in a single line, something like:

`results = get_all_members.select {|member| member.search_key === search_key}`

Reply:
Yep! But note that in Ruby, `===` is not actually recommended for use as an equality operator. Also, I could use a review on https://github.com/code-dot-org/code-dot-org/pull/48989 when folks have a chance!